src/nlp/api.jl

```diff
@@ -11,6 +11,7 @@ export jth_hess_coord, jth_hess_coord!, jth_hess
 export jth_hprod, jth_hprod!, ghjvprod, ghjvprod!
 export hess_structure!, hess_structure, hess_coord!, hess_coord
 export hess, hprod, hprod!, hess_op, hess_op!
+export tensor_projection, tensor_projection!
```

Comment:
What is your testing plan?
Update NLPModelsTest.jl would be an option, I think:
- BROWNDEN, HS13, HS5
would qualify.

Reply:
For now, we try to make it working with ForwardDiff.jl and Enzyme.jl and check that we can do that with AD such that the feature is relevant.
For Enzyme.jl, we need to wait next month and the help of Valentin but it should work for ForwardDiff.jl with multiple "forward over forward".

I will add some tests in `NLPModelsTest.jl` next week if it works as expected with `ForwardDiff.jl`.

Reply:
Looks good, thanks.